glib/src/gstring.rs

```diff
@@ -989,12 +1014,15 @@ impl IntoGlibPtr<*mut c_char> for GString {
     /// Transform into a nul-terminated raw C string pointer.
     unsafe fn into_glib_ptr(self) -> *mut c_char {
         match self.0 {
-            Inner::Native(None) => ffi::g_malloc0(1) as *mut _,
+            Inner::Native(None) => unreachable!(),
```

Comment:
As this case is impossible now, maybe we can get rid of the `Option` instead?

Reply:
Removed the `Option`, any time we need to take the `Box` it uses `mem::replace(b, "".into())` which always gets optimized to avoid allocation